pkg/test/client.go

```diff
@@ -16,13 +16,14 @@ import (
 )
 
 // NewFakeClient creates a fake K8s client with ability to override specific Get/List/Create/Update/StatusUpdate/Delete functions
-func NewFakeClient(t T, initObjs ...runtime.Object) *FakeClient {
+func NewFakeClient(t T, initObjs ...client.Object) *FakeClient {
```

Comment:
changes initObjs type from runtime.Object to client.Object - see discussion [here](https://redhat-internal.slack.com/archives/CHK0J6HT6/p1709236452856659)
WithStatusSubresource can only accept client.Object was a major reason to consider making this change.